.github/workflows/benchmarks_run.yml

```diff
@@ -0,0 +1,40 @@
+name: Run and Cache Benchmarks
+
+on:
+  pull_request:
+    types: [ opened, reopened, edited, synchronize ]
+
+jobs:
+  benchmark:
+    name: Run benchmarks
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout main branch
+        uses: actions/checkout@v2
+        with:
+          ref: main
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@v2
+
+      - name: Run benchmarks on main - solvers
+        run: cargo bench -- solvers --exact | tail -n +6  > results_main.txt
+
+      - name: Checkout PR branch
+        uses: actions/checkout@v2
+
+      - name: Run benchmarks on PR branch
+        run: cargo bench -- solvers --exact | tail -n +6 > results_pr.txt
```

Comment:
If the two benchmarks are run in separate `jobs` they can run in parallel which should speed things up. Currently the whole workflow runs in 55s which seems plenty fast, so maybe that is an unnecessary optimization, but I don't know how much slower things are expected to get in the future.

Reply:
My thought there was to try to minimizie the chance that a difference would appear because the jobs were scheduled on machines with different performance characteristics. But that is good to keep in mind.